### PR TITLE
Notification improvements

### PR DESCRIPTION
**Pull request by WiserTixx**
Adds icons to ui notifications/messages and adds a category to all ui notifications to avoid them overwriting another. Also adds a new packet "N" which allows for custom notifications with category and icon

### CODE REVIEW

File: lua/ge/extensions/MPGameNetwork.lua [hunk 1]
```diff
@@ -265,13 +265,14 @@ local HandleNetwork = {
 	['Z'] = function(params) positionGE.handle(params) end, -- position and velocity
 	['O'] = function(params) MPVehicleGE.handle(params) end, -- all vehicle spawn, modification and delete events, couplers
 	['P'] = function(params) MPConfig.setPlayerServerID(params) end,
-	['J'] = function(params) MPUpdatesGE.onPlayerConnect() UI.showNotification(params) end, -- A player joined
+	['J'] = function(params) MPUpdatesGE.onPlayerConnect() UI.showNotification(params,nil,"person_add") end, -- A player joined
 	['L'] = function(params) UI.showNotification(params) end, -- Display custom notification
```
Comment (20dka) [inline lua/ge/extensions/MPGameNetwork.lua]: please update the comment to signal that this packet is only used for Leaving users
maybe even extract the string matching related to this from `showNotification`, to its own function

File: lua/ge/extensions/MPGameNetwork.lua [hunk 2]
```diff
@@ -265,13 +265,14 @@ local HandleNetwork = {
...
 	['S'] = function(params) sessionData(params) end, -- Update Session Data
 	['E'] = function(params) handleEvents(params) end, -- Event For another Resource
 	['T'] = function(params) quitMP(params) end, -- Player Kicked Event (old, doesn't contain reason)
 	['K'] = function(params) quitMP(params) end, -- Player Kicked Event (new, contains reason)
 	['C'] = function(params) UI.chatMessage(params) end, -- Chat Message Event
+	['N'] = function(params) local category, icon, message = packet:match("([^:]+):([^:]+):(.+)") UI.showNotification(message, category, icon) end,
```
Comment (20dka) [inline lua/ge/extensions/MPGameNetwork.lua]: please add a comment describing the packet type
```suggestion
	['N'] = function(params) local category, icon, message = packet:match("([^:]+):([^:]+):(.+)") UI.showNotification(message, category, icon) end, -- Custom UI notification
```

File: lua/ge/extensions/MPVehicleGE.lua [hunk 3]
```diff
@@ -1171,7 +1171,7 @@ local function spawnDestroyedVehicles(serverVehID)
     end
 
     if vehicles[serverVehID].isDeleted == false then
-        UI.showNotification('This vehicle hasnt been deleted yet?')
+        UI.showNotification(''..vehicles[serverVehID].ownerName..'\'s '..vehicles[serverVehID].jbeam..' hasnt been deleted yet?', ''..serverVehID..'delete', 'warning')
```
Comment (20dka) [inline lua/ge/extensions/MPVehicleGE.lua]: please fix the typo "hasnt" -> "hasn't"

File: lua/ge/extensions/MPVehicleGE.lua [hunk 4]
```diff
@@ -1194,7 +1194,7 @@ local function spawnDestroyedVehicles(serverVehID)
         serverVehicleID = serverVehID,
         data = encodedVehicleData
     }
-    UI.showNotification('Trying to respawn '..playerOwnerName)
+    UI.showNotification('Trying to respawn '..playerOwnerName..'\'s '..vehicles[serverVehID].jbeam, ''..playerOwnerName..''..serverVehID..'respawn', 'directions_car')
```
Comment (20dka) [inline lua/ge/extensions/MPVehicleGE.lua]: optional: please switch the string to `""` if it contains a `'` character, using `\'` is not preferred
(repeat where applicable)

File: lua/ge/extensions/UI.lua [hunk 5]
```diff
@@ -195,18 +195,17 @@ end
...
-		if leftName then MPVehicleGE.onPlayerLeft(leftName) end
-		--local joinedName = string.match(text, "^Welcome (.+)!$")
-		--if joinedName then MPVehicleGE.onPlayerJoined(joinedName) end
+-- @param category string 
+-- @param icon string material_ icons from ui\assets\Sprites\svg-symbols.svg example: smoking_rooms
+local function showNotification(text, category, icon)
+	log('I', 'showNotification', "[Message] > "..tostring(text))
+	local leftName = string.match(text, "^(.+) left the server!$")
+	if leftName then 
+		MPVehicleGE.onPlayerLeft(leftName)
```
Comment (20dka) [inline lua/ge/extensions/UI.lua]: consider extracting this section from this function, since we know it comes from a single source, the `L` packet

**Review by 20dka — CHANGES_REQUESTED** [review]

**Review by Starystars67 — DISMISSED** [review]
LGTM though not tested.

**Review by Starystars67 — DISMISSED** [review]

**Review by 20dka — APPROVED** [review]

**Pull request merged**